File: java/686.java
class Solution {
    public int repeatedStringMatch(String A, String B) {
        int l = A.length(),k = B.length();
        int i = 1;
        String s = "";
        while(i <= k/l+2){
            s += A;
            if(s.lastIndexOf(B) != -1){
                return i;
            }
            i ++;
        }
        
        
        return -1;
    }
}

File: go/686.go
import "strings"

func repeatedStringMatch(A string, B string) int {
	S := ""

	for i := 1; i <= len(B)/len(A)+2; i++ {
		S = S + A
		if strings.Contains(S, B) {
			return i
		}
	}
	return -1
}